src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -667,60 +631,70 @@ private static void FlattenDeconstructVariables(ArrayBuilder<DeconstructionVaria
         /// </summary>
-        private DeconstructionVariable BindDeconstructionDeclarationVariables(
-            ExpressionSyntax node,
-            DiagnosticBag diagnostics)
+        private DeconstructionVariable BindDeconstructionVariables(ExpressionSyntax node, DiagnosticBag diagnostics)
         {
             switch (node.Kind())
             {
                 case SyntaxKind.DeclarationExpression:
                     {
                         var component = (DeclarationExpressionSyntax)node;
-                        return BindDeconstructionDeclarationVariables(component.Type, component.Designation, diagnostics);
+                        return BindDeconstructionVariables(component.Type, component.Designation, diagnostics);
                     }
                 case SyntaxKind.TupleExpression:
                     {
                         var component = (TupleExpressionSyntax)node;
                         var builder = ArrayBuilder<DeconstructionVariable>.GetInstance(component.Arguments.Count);
                         foreach (var arg in component.Arguments)
                         {
-                            builder.Add(BindDeconstructionDeclarationVariables(arg.Expression, diagnostics));
+                            builder.Add(BindDeconstructionVariables(arg.Expression, diagnostics));
                         }
                         return new DeconstructionVariable(builder, node);
                     }
                 case SyntaxKind.IdentifierName:
                     {
                         var identifier = (IdentifierNameSyntax)node;
-                        if (identifier.Identifier.ContextualKind() == SyntaxKind.UnderscoreToken)
+                        if (identifier.Identifier.ContextualKind() != SyntaxKind.UnderscoreToken)
                         {
-                            return new DeconstructionVariable(new BoundDiscardedExpression(identifier, type: null, hasErrors: false), node);
+                            goto default;
                         }
 
-                        // TODO REVIEW: handle expressions, not just declarations
-                        goto default;
+                        // PROTOTYPE(wildcards): is this the right lookup?
+                        var name = identifier.Identifier.ValueText;
+                        LookupOptions options = LookupOptions.AllMethodsOnArityZero;
+                        var lookupResult = LookupResult.GetInstance();
+                        HashSet<DiagnosticInfo> useSiteDiagnostics = null;
+                        this.LookupSymbolsWithFallback(lookupResult, name, arity: 0,
+                                useSiteDiagnostics: ref useSiteDiagnostics, options: options);
+                        diagnostics.Add(node, useSiteDiagnostics);
+
+                        // If the lookup is successful, bind as expression, other as discard
+                        if (lookupResult.Kind != LookupResultKind.Empty)
```

Comment:
Inaccessible symbols should be treated as not found as well. I think you should try to maximize reuse of the normal binding path.